django/conf/global_settings.py

```diff
@@ -492,6 +492,7 @@
 PASSWORD_HASHERS = [
     'django.contrib.auth.hashers.PBKDF2PasswordHasher',
     'django.contrib.auth.hashers.PBKDF2SHA1PasswordHasher',
+    'django.contrib.auth.hashers.Argon2PasswordHasher',
```

Comment:
1.  On compatibility.   The hasher is not as compatible as PBKDF2, but it is close.  `argon2-cffi`, the library used in the PR, comes with `libargon2` bundled with which it binds with cffi instead of ctypes.  This means it will work just fine on lots of platforms including pypy.  It will exclude, for instance, Google App Engine.
2. On security.  The panel of the Password Hashing Competition recommends the use of argon2 over PBKDF2, which it considers a legacy algorithm.   The only downside I see, is its youth.

I think it would be great to have argon2 in Django as an option, such that those who are wary of PBKDF2 can switch to argon2 with minimal effort.

Maybe we could switch to argon2 as default in a later release.  Some questions:

1. For ease of deployment, `argon2-cffi` could be bundled with Django?
2. Wouldn't it be possible to make `get_hasher` find the first available hasher, to support older platforms.
3. I could write a pure python version of `argon2-cffi` that only implements verification, such that platforms that become incompatible can move away to another hasher?